flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannelTest.java

```diff
@@ -456,6 +463,23 @@ public void testUnblockReleasedChannel() throws Exception {
 		localChannel.resumeConsumption();
 	}
 
+	@Test
+	public void testNoNotifyOnSavepoint() throws IOException {
+		TestBufferReceivedListener listener = new TestBufferReceivedListener();
+		LocalInputChannel channel = new LocalInputChannel(
+			new SingleInputGateBuilder().build(),
+			0,
+			new ResultPartitionID(),
+			new ResultPartitionManager(),
+			new TaskEventDispatcher(),
+			new TestCounter(),
+			new TestCounter());
+		CheckpointBarrier barrier = new CheckpointBarrier(123L, 123L, new CheckpointOptions(SAVEPOINT, CheckpointStorageLocationReference.getDefault()));
+		channel.notifyPriorityEvent(new BufferConsumer(toBuffer(barrier).getMemorySegment(), FreeingBufferRecycler.INSTANCE, getDataType(barrier)));
+		channel.checkError();
+		assertTrue(listener.notifiedOnBarriers.isEmpty());
```

Comment:
frankly, it's not that unexpected that it's empty, since `listener` was never passed anywhere? 😈 

(you forgot to register it to `SingleInputGate`?)